openedx/adg/lms/applications/helpers.py

```diff
@@ -865,3 +872,40 @@ def send_application_status_update_email(application, message_for_applicant):
     task_send_mandrill_email.delay(
         status_email_template_map[application.status], [applicant.email], email_context
     )
+
+
+def check_and_update_prereq_courses_status_to_completed(user_id, course_id):
+    """
+    Update prereq_courses_status to COMPLETED in Application Hub
+    depending on the prereq courses that the learner has enrolled in
+    """
+    from openedx.adg.lms.applications.models import ApplicationHub, MultilingualCourseGroup
+
+    current_user = User.objects.get(id=user_id)
+    hub = ApplicationHub.objects.filter(user=current_user).first()
+    is_any_status_updated = False
```

Comment:
No need to keep this flag. Once we find out that a course is program or bu pre req and is also completed then just simply update the hub and save it. A cannot be bu pre req and a program pre req at the same time.